crates/ty_project/src/db/changes.rs

```diff
@@ -138,7 +175,9 @@ impl ProjectDatabase {
                     };
 
                     if is_file {
-                        sync_path(self, &path);
+                        if synced_files.insert(path.to_path_buf()) {
+                            File::sync_path(self, &path);
+                        }
```

Comment:
On line 284: We'll need to re-create the `FileRoot`. The case handled there is if a user deletes a `pyproject.toml` (or creates a new `ty.toml`). Doing so can change the root path of the project. You might want to move the `FileRoot` creation into the `from_metadata` method